File: Data Structures & Algorithms/find-duplicate-integer/submission-2.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        for index, val in enumerate(nums):
            num = nums[index]
            
            if nums[abs(num) - 1] < 0:
                return abs(num)
            
            nums[abs(num) - 1] *= -1

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int res{};
        unordered_set<int> seen;

        for(const auto& num: nums){
            seen.insert(num);
        }

        for(const auto& num:seen){

            if(!seen.count(num - 1)){
                int search_num = num;
                int lenght = 0;
                while(seen.count(search_num)){
                    search_num++;
                    lenght++;
                }
                res = max(res, lenght);
            }

        }

        return res;
    }
};
